src/MongoDB.Driver/LibMongoCrypt/LibMongoCryptController.cs

```diff
@@ -0,0 +1,415 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Crypt;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.WireProtocol;
+
+namespace MongoDB.Driver.LibMongoCrypt
+{
+    internal class LibMongoCryptController : IBinaryDocumentFieldDecryptor, IBinaryDocumentFieldEncryptor//, IBinaryExplicitFieldEncryption
```

Comment:
Probably we need to add a new interface for `explicit` encryption or add methods `explicit` encryption methods in `IBinaryDocumentFieldDecryptor` and `IBinaryDocumentFieldEncryptor`